.github/workflows/build-and-test.yml

```diff
@@ -20,12 +20,12 @@ jobs:
           ref: ${{ github.event.pull_request.head.sha }}
 
       - name: Install pnpm
-        run: npm i pnpm@8.14.1 --global
+        run: npm install -g pnpm@8.14.0
```

Comment:
Was wondering why this was downgrading, but I see you're using a consistent version throughout below.

Question: why are we using such an old version (8.14.0 is over a year ago, latest 8.x is 8.15.8 from last April, I think). It might be worth testing if everything works with the latest 8.x version (and later see if we can upgrade to latest major version).

Nice to use long args rather than short ones (fine either way, but would use `--global` throughout for the same reason that `install` is used rather than `i`)